### PR TITLE
Module integration bugfixes

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description
Added timer on GitOps, notifier and Security module so that the nav option will be visible after the installation without navigating to somewhere else or reload

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Manually tested

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/devtronStackManager/DevtronStackManager.tsx [hunk 1]
```diff
@@ -274,6 +279,12 @@ export default function DevtronStackManager({
                             }
                             _installedModulesList.push(_moduleDetails)
                         }
+                        if(moduleInInstallingState === _moduleDetails.name && result.status !== ModuleStatus.INSTALLING){
+                          setModuleInInstallingState('')
+                        }
+                        if (result.status === ModuleStatus.INSTALLING) {
```
Comment (sohel-rp) [inline src/components/v2/devtronStackManager/DevtronStackManager.tsx:285]: Please use else if

File: src/components/v2/devtronStackManager/DevtronStackManager.tsx [hunk 2]
```diff
@@ -190,6 +190,11 @@ export default function DevtronStackManager({
                         ...currentModule,
                         installationStatus: result.status,
                     })
+                    if(moduleInInstallingState === result.name && result.status !== ModuleStatus.INSTALLING){
```
Comment (shivani170) [inline src/components/v2/devtronStackManager/DevtronStackManager.tsx:193]: Repeating code line. Can we optimize this function

File: src/components/globalConfigurations/GlobalConfiguration.tsx [hunk 3]
```diff
@@ -200,26 +200,42 @@ function NavItem({ hostURLConfig, serverMode }) {
...
             if (result?.status === ModuleStatus.INSTALLED) {
                 setInstalledModule([...installedModule, ModuleNameMap.NOTIFICATION])
+            } else if (result?.status === ModuleStatus.INSTALLING) {
+                moduleStatusTimer = setTimeout(() => {
+                  getNotificationModuleStatus(3)
+                }, 15000)
             }
-        } catch (error) {}
+        } catch (error) {
+            if (retryOnError >= 0) {
```
Comment (shivani170) [inline src/components/globalConfigurations/GlobalConfiguration.tsx:235]: Will there be any condition where `retryOnError < 0` if yes how are we handling that

Comment (vivek-devtron) [reply src/components/globalConfigurations/GlobalConfiguration.tsx:235]: No

File: src/components/globalConfigurations/GlobalConfiguration.tsx [hunk 4]
```diff
@@ -200,26 +200,42 @@ function NavItem({ hostURLConfig, serverMode }) {
...
 
-    const getNotificationModuleStatus = async (): Promise<void> => {
+    const getNotificationModuleStatus = async (retryOnError: number): Promise<void> => {
         try {
             const { result } = await getModuleInfo(ModuleNameMap.NOTIFICATION)
             if (result?.status === ModuleStatus.INSTALLED) {
                 setInstalledModule([...installedModule, ModuleNameMap.NOTIFICATION])
+            } else if (result?.status === ModuleStatus.INSTALLING) {
+                moduleStatusTimer = setTimeout(() => {
+                  getNotificationModuleStatus(3)
```
Comment (shivani170) [inline src/components/globalConfigurations/GlobalConfiguration.tsx:231]: can we keep these number like 0,3 in constant

File: src/components/globalConfigurations/GlobalConfiguration.tsx [hunk 5]
```diff
@@ -200,26 +200,42 @@ function NavItem({ hostURLConfig, serverMode }) {
...
     }, [])
 
-    async function getGitOpsModuleStatus() {
+    const getGitOpsModuleStatus = async (retryOnError: number): Promise<void> => {
         try {
             const { result } = await getModuleInfo(ModuleNameMap.ARGO_CD)
             if (result?.status === ModuleStatus.INSTALLED) {
                 setInstalledModule([...installedModule, ModuleNameMap.ARGO_CD])
+            } else if (result?.status === ModuleStatus.INSTALLING) {
+                moduleStatusTimer = setTimeout(() => {
```
Comment (shivani170) [inline src/components/globalConfigurations/GlobalConfiguration.tsx:213]: Repetitive code like below

Comment (vivek-devtron) [reply src/components/globalConfigurations/GlobalConfiguration.tsx:213]: Done

File: src/components/globalConfigurations/GlobalConfiguration.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/globalConfigurations/GlobalConfiguration.tsx]: Done

File: src/components/v2/devtronStackManager/DevtronStackManager.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/v2/devtronStackManager/DevtronStackManager.tsx]: Done

File: src/components/common/navigation/Navigation.tsx [hunk 6]
```diff
@@ -136,8 +146,8 @@ export default class Navigation extends Component<
                 }))
             } else if (result?.status === ModuleStatus.INSTALLING) {
                 this.securityModuleStatusTimer = setTimeout(() => {
-                    this.getSecurityModuleStatus(3)
-                }, 30000)
+                    this.getSecurityModuleStatus(MODULE_STATUS_RETRY_COUNT)
+                }, MODULE_STATUS_RETRY_COUNT)
```
Comment (sohel-rp) [inline src/components/common/navigation/Navigation.tsx:150]: This should be `MODULE_STATUS_POLLING_INTERVAL`

**Review by shivani170 — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Review by sohel-rp — APPROVED** [review]

**Review by shivani170 — APPROVED** [review]

**Pull request merged**